File: javascript/easy/palindromeNumber.js
let isPalindrome = function(x) {
    return x < 0 ? false : (reverse(x, {f: 1}) == x);
};

let reverse = function(n, obj) {
    if(n <= 0) return 0;
    let val = reverse(parseInt(n/10), obj);
    res = val + (n%10) * obj.f;
    obj.f *= 10;
    return res;
};

File: ruby/easy/palindrome_number.rb
def is_palindrome(x)
  return (x >= 0) && (reverse(x) == x)
end

def reverse(x)
    res = 0
    while(x > 0)
      res = (res*10) + (x%10)
      x = (x/10).to_i
    end
    return res
end